test/controllers/MessagesControllerSpec.scala

```diff
@@ -96,26 +103,36 @@ class MessagesControllerSpec extends AnyWordSpec with Matchers with GuiceOneAppP
   private val serviceConfig = new ServicesConfig(configuration)
   private val appConfig     = new AppConfig(configuration, serviceConfig)
 
-  private val controller = new MessagesController(appConfig, Helpers.stubControllerComponents())
+  private def controller(messageConnector: MessageConnector = mock[MessageConnector]) = new MessagesController(appConfig, Helpers.stubControllerComponents(), messageConnector)
 
   "POST any XML" should {
-    "should return 202 Accepted" in {
-      val result = controller.post()(fakeValidXmlRequest)
+    "should return 202 Accepted when message connector successful" in {
+      val mc = mock[MessageConnector]
+      when(mc.post(any())(any())).thenReturn(Future.successful(HttpResponse(ACCEPTED, "")))
+
+      val result = controller(mc).post()(fakeValidXmlRequest)
       status(result) shouldBe ACCEPTED
     }
+    "should return 500 Internal Server Error when message connector receives 500" in {
```

Comment:
More tests could be added for the recoded status codes.

Reply:
Thats primarily covered by the CustomHttpReaderSpec.